### PR TITLE
feat: Add `LinkTarget` to `FileSystemInfo` for .NET 6 targets

### PR DESCRIPTION
**Pull request by BrianMcBrayer**
## Overview

In .NET 6, additional symlinks support was added to `FileSystemInfo`. This PR adds at least one of those properties here, the `LinkTarget` property.

See https://github.com/TestableIO/System.IO.Abstractions/issues/789

We need to discuss the mock portion of this though as I'm not sure how this codebase expects that.

### CODE REVIEW

**Comment by BrianMcBrayer** [comment]
I think that this is probably a minor version bump, since it adds to the API without changing existing APIs? But I'm not sure.

File: src/System.IO.Abstractions.TestingHelpers/MockDirectoryInfo.cs
Comment (BrianMcBrayer) [inline src/System.IO.Abstractions.TestingHelpers/MockDirectoryInfo.cs:138]: I need guidance on this. Where would we set/get this data? I'll take a look tonight to see, but if someone can point me in the right direction is might save some time.

File: src/System.IO.Abstractions.TestingHelpers/MockFileInfo.cs
Comment (BrianMcBrayer) [inline src/System.IO.Abstractions.TestingHelpers/MockFileInfo.cs:185]: I need guidance on this. Where would we set/get this data? I'll take a look tonight to see, but if someone can point me in the right direction is might save some time.

File: src/System.IO.Abstractions.TestingHelpers/MockDirectoryInfo.cs
Comment (fgreinacher) [inline src/System.IO.Abstractions.TestingHelpers/MockDirectoryInfo.cs:134]: We currently use dedicated "per-feature" constants for such things (see https://github.com/TestableIO/System.IO.Abstractions/blob/main/Directory.Build.props#L13-L15). Would you please follow the same pattern (just for consistency)?

File: src/System.IO.Abstractions.TestingHelpers/MockDirectoryInfo.cs
Comment (fgreinacher) [reply src/System.IO.Abstractions.TestingHelpers/MockDirectoryInfo.cs:138]: I see two options:

1. Keep it as is, returning a fixed `null`, and implement the mock part later
2. Allow defining link targets in [`MockFileData`](https://github.com/TestableIO/System.IO.Abstractions/blob/d7ba4dc9a47ec054730aba16fb1995ad77809cea/src/System.IO.Abstractions.TestingHelpers/MockFileData.cs)/[`MockDirectoryData`](https://github.com/TestableIO/System.IO.Abstractions/blob/d7ba4dc9a47ec054730aba16fb1995ad77809cea/src/System.IO.Abstractions.TestingHelpers/MockDirectoryData.cs)

I'd slightly prefer option `2` because I'm pretty sure people will quickly want to mock the new property, but I'm also totally fine doing that later 😀

File: src/System.IO.Abstractions.TestingHelpers/MockFileInfo.cs
Comment (fgreinacher) [reply src/System.IO.Abstractions.TestingHelpers/MockFileInfo.cs:185]: Same comments as above.

File: src/System.IO.Abstractions.TestingHelpers/System.IO.Abstractions.TestingHelpers.csproj
Comment (fgreinacher) [inline src/System.IO.Abstractions.TestingHelpers/System.IO.Abstractions.TestingHelpers.csproj:6]: OMG, I should actually have done this in https://github.com/TestableIO/System.IO.Abstractions/pull/777/files 😬 Thanks!

File: src/System.IO.Abstractions/DirectoryInfoWrapper.cs
Comment (fgreinacher) [inline src/System.IO.Abstractions/DirectoryInfoWrapper.cs:99]: Same as above: Please use "per-feature" constants

File: src/System.IO.Abstractions/FileInfoWrapper.cs
Comment (fgreinacher) [inline src/System.IO.Abstractions/FileInfoWrapper.cs:97]: Same as above: Please use "per-feature" constants

File: src/System.IO.Abstractions/FileSystemInfoBase.cs
Comment (fgreinacher) [inline src/System.IO.Abstractions/FileSystemInfoBase.cs:57]: Same as above: Please use "per-feature" constants

File: tests/System.IO.Abstractions.TestingHelpers.Tests/MockDirectoryArgumentPathTests.cs
Comment (fgreinacher) [inline tests/System.IO.Abstractions.TestingHelpers.Tests/MockDirectoryArgumentPathTests.cs:31]: 🏅

File: src/System.IO.Abstractions.TestingHelpers/MockFileInfo.cs
Comment (fgreinacher) [inline src/System.IO.Abstractions.TestingHelpers/MockFileInfo.cs:181]: Same as above: Please use "per-feature" constants

**Review by fgreinacher — CHANGES_REQUESTED** [review]
Thanks a lot @BrianMcBrayer! Left some comments inline

**Comment by BrianMcBrayer** [comment]
Thank you! I'll try to update it all today

File: src/System.IO.Abstractions.TestingHelpers/MockDirectoryInfo.cs
Comment (BrianMcBrayer) [reply src/System.IO.Abstractions.TestingHelpers/MockDirectoryInfo.cs]: Updated

File: src/System.IO.Abstractions.TestingHelpers/MockDirectoryInfo.cs
Comment (BrianMcBrayer) [reply src/System.IO.Abstractions.TestingHelpers/MockDirectoryInfo.cs]: Updated, but I'll wait until #791 goes in and then I'll update `MockFileInfo` again to match that updated pattern.

File: src/System.IO.Abstractions.TestingHelpers/MockFileInfo.cs
Comment (BrianMcBrayer) [reply src/System.IO.Abstractions.TestingHelpers/MockFileInfo.cs]: Done

File: src/System.IO.Abstractions.TestingHelpers/MockFileInfo.cs
Comment (BrianMcBrayer) [reply src/System.IO.Abstractions.TestingHelpers/MockFileInfo.cs]: Done

File: src/System.IO.Abstractions/DirectoryInfoWrapper.cs
Comment (BrianMcBrayer) [reply src/System.IO.Abstractions/DirectoryInfoWrapper.cs]: Done

File: src/System.IO.Abstractions/FileInfoWrapper.cs
Comment (BrianMcBrayer) [reply src/System.IO.Abstractions/FileInfoWrapper.cs]: Done

File: src/System.IO.Abstractions/FileSystemInfoBase.cs
Comment (BrianMcBrayer) [reply src/System.IO.Abstractions/FileSystemInfoBase.cs]: Done

**Review by BrianMcBrayer — COMMENTED** [review]
I think I addressed all the comments

**Comment by BrianMcBrayer** [comment]
Note that this PR is pending #791 because of the enhancements that will add to `MockFileInfo`

**Review by fgreinacher — APPROVED** [review]
Looking very good @BrianMcBrayer 👍

Happy to merge after you rebased/adapted the Mock implementation!

File: Directory.Build.props
Comment (fgreinacher) [inline Directory.Build.props:16]:
```suggestion
    <DefineConstants Condition="'$(TargetFramework)' == 'net6.0' OR '$(TargetFramework)' == 'net5.0'">$(DefineConstants);FEATURE_FILE_MOVE_WITH_OVERWRITE;FEATURE_SUPPORTED_OS_ATTRIBUTE;FEATURE_FILE_SYSTEM_WATCHER_FILTERS;FEATURE_ENDS_IN_DIRECTORY_SEPARATOR;FEATURE_PATH_JOIN_WITH_PARAMS;FEATURE_PATH_JOIN_WITH_FOUR_PATHS</DefineConstants>
    <DefineConstants Condition="'$(TargetFramework)' == 'net6.0'">$(DefineConstants);FEATURE_FILE_SYSTEM_INFO_LINK_TARGET</DefineConstants>
```

File: src/System.IO.Abstractions.TestingHelpers/MockDirectoryInfo.cs
Comment (fgreinacher) [inline src/System.IO.Abstractions.TestingHelpers/MockDirectoryInfo.cs:146]:
```suggestion
        /// <inheritdoc />
        public override string Name
```

File: src/System.IO.Abstractions.TestingHelpers/MockFileData.cs
Comment (fgreinacher) [inline src/System.IO.Abstractions.TestingHelpers/MockFileData.cs:145]:
```suggestion
        public string LinkTarget { get; set; }
```

File: src/System.IO.Abstractions.TestingHelpers/MockFileInfo.cs
Comment (fgreinacher) [inline src/System.IO.Abstractions.TestingHelpers/MockFileInfo.cs:176]: I guess this just needs to read:

```suggestion
                var mockFileData = GetMockFileDataForRead();
```

**Review by fgreinacher — CHANGES_REQUESTED** [review]
@BrianMcBrayer Thanks again, I left some minor suggestions. Please also bump the minor version in https://github.com/BrianMcBrayer/System.IO.Abstractions/blob/bug/add-link-target-file-system-info/version.json#L3 to indicated the new feature.

File: src/System.IO.Abstractions.TestingHelpers/MockDirectoryInfo.cs
Comment (BrianMcBrayer) [reply src/System.IO.Abstractions.TestingHelpers/MockDirectoryInfo.cs:146]: Oh wow. I'm sorry about this. Not even sure how it happened :man_facepalming:

File: src/System.IO.Abstractions.TestingHelpers/MockFileInfo.cs
Comment (BrianMcBrayer) [reply src/System.IO.Abstractions.TestingHelpers/MockFileInfo.cs:176]: :man_facepalming:

**Comment by BrianMcBrayer** [comment]
@fgreinacher I accepted your review comments. Can't believe I didn't indent those things correctly. I think the conditional comments threw me off, which is silly.

**Review by fgreinacher — CHANGES_REQUESTED** [review]
Thanks a ton @BrianMcBrayer and no worries about the minor style issues.

Please bump the minor version in https://github.com/BrianMcBrayer/System.IO.Abstractions/blob/bug/add-link-target-file-system-info/version.json#L3 to indicate the new feature. I would have done this myself but cannot commit to your branch (you can allow that by ticking the "Allow edits and access to secrets by maintainers" checkbox on the right).

**Review by fgreinacher — APPROVED** [review]
Looking great now, thanks for addressing everything so quickly!

**Pull request merged**
